File: app/src/main/java/org/nitri/opentopo/model/LocationViewModel.java
package org.nitri.opentopo.model;

import android.location.Location;

import androidx.lifecycle.MutableLiveData;
import androidx.lifecycle.ViewModel;

public class LocationViewModel extends ViewModel {

    private MutableLiveData<Location> currentLocation;
    private MutableLiveData<String> nmea;

    public MutableLiveData<Location> getCurrentLocation() {
        if (currentLocation == null) {
            currentLocation = new MutableLiveData<>();
        }
        return currentLocation;
    }

    public MutableLiveData<String> getCurrentNmea() {
        if (nmea == null) {
            nmea = new MutableLiveData<>();
        }
        return nmea;
    }

}

File: app/src/main/java/org/nitri/opentopo/model/LocationViewModel.kt
package org.nitri.opentopo.model

import android.location.Location
import androidx.lifecycle.MutableLiveData
import androidx.lifecycle.ViewModel

class LocationViewModel : ViewModel() {
    var currentLocation: MutableLiveData<Location>? = null
        get() {
            if (field == null) {
                field = MutableLiveData()
            }
            return field
        }
        private set
    private var nmea: MutableLiveData<String>? = null
    val currentNmea: MutableLiveData<String>
        get() {
            if (nmea == null) {
                nmea = MutableLiveData()
            }
            return nmea as MutableLiveData<String>
        }
}
